File: company/DataSerializable.java
package com.company;

import java.io.*;

public class DataSerializable<E extends Comparable<E>> {

    public void serializeToBinary(myList<E> data, String filename) throws IOException {
        ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(filename));
        oos.writeObject(data);
        oos.close();
    }

    public static Object deserializeFromBinary(String filename) throws IOException, ClassNotFoundException {
        ObjectInputStream ois = new ObjectInputStream(new FileInputStream(filename));
        Object data = ois.readObject();
        ois.close();
        return data;
    }

}

File: src/main/scala/com/ScalaLR/scala/DataSerializable.scala
package com.ScalaLR.scala

import java.io._

class DataSerializable[E <: Comparable[E]] {
  def serializeToBinary(data: myList[E], filename: String): Unit = {
    val oos = new ObjectOutputStream(new FileOutputStream(filename))
    oos.writeObject(data)
    oos.close()
  }

  def deserializeFromBinary(filename: String): Any = {
    val ois = new ObjectInputStream(new FileInputStream(filename))
    val data = ois.readObject()
    ois.close()
    data
  }
}
